plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/CreateAppEngineWtpProjectTest.java

```diff
@@ -143,22 +141,17 @@ private boolean containsSegment(IPath path, String segment) {
 
   @Test
   public void testNoTestClassesInDeploymentAssembly()
-      throws InvocationTargetException, CoreException {
+      throws InvocationTargetException, CoreException, InterruptedException {
     CreateAppEngineWtpProject creator = newCreateAppEngineWtpProject();
     creator.execute(monitor);
 
-    assertNoTestClassesInDeploymentAssembly();
-  }
-
-  private void assertNoTestClassesInDeploymentAssembly() {
-    StructureEdit core = StructureEdit.getStructureEditForRead(project);
-    try {
-      WorkbenchComponent component = core.getComponent();
-      assertEquals(0, component.findResourcesBySourcePath(new Path("/src/test/java"), 0).length);
-      assertEquals(1, component.findResourcesBySourcePath(new Path("/src/main/java"), 0).length);
-    } finally {
-      core.dispose();
+    while (!creator.deployAssemblyEntryRemoveJob.jobCompleted) {
+      Thread.sleep(50);
```

Comment:
It'll be killed by Tycho's test timeout.  But could use a `Stopwatch` from Guava to keep track.

I think a comment here re: NonSystemJobSuspender would be helpful.